Updating valid .NET SDK targets for Quickstart docs

WhitWaldo opened the pull request:
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [X] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [X] Commands include options for Linux, MacOS, and Windows within codetabs
- [X] New file and folder names are globally unique
- [X] Page references use shortcodes instead of markdown or URL links
- [X] Images use HTML style and have alternative text
- [X] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Updated documentation for .NET Quickstarts to reflect that .NET 6, .NET 8 and .NET 9 are valid SDK targets, but also added standard deprecation warning indicating that only .NET 8 and .NET 9 will be supported in v1.16.

**Please do not merge until https://github.com/dapr/dotnet-sdk/pull/1404 has been merged**

WhitWaldo committed:
Updated all Quickstart docs to reflect .NET 9 as a valid target + added 1.16 deprecation warnings for .NET 6 and .NET 7

Signed-off-by: Whit Waldo <whit.waldo@innovian.net>

msfussell commented on daprdocs/content/en/getting-started/quickstarts/configuration-quickstart.md:
```diff
@@ -272,10 +272,18 @@ setTimeout(() => {
+Note that while .NET 6 is generally supported as the minimum .NET requirement across the Dapr .NET SDK packages
+and .NET 7 is the minimally supported version of .NET by Dapr.Workflows in Dapr v1.15, only .NET 8 and .NET 9 wil
+continue to be supported by Dapr in v1.16 and later.
+
```
I would make this more succient as below. Also the comment on Workflows only need to be added to the WF example. I suggest

.NET 6 is supported as the minimum requirement for the Dapr .NET SDK packages in this release.
Only .NET 8 and .NET 9 will be supported in Dapr v1.16 and later releases.

msfussell commented on daprdocs/content/en/getting-started/quickstarts/actors-quickstart.md:
```diff
@@ -30,10 +30,18 @@ As a quick overview of the .NET actors quickstart:
+
+{{% alert title="Note" color="primary" %}}
+
+Note that while .NET 6 is generally supported as the minimum .NET requirement across the Dapr .NET SDK packages
```
I would make this more succient as below. Also the comment on Workflows only need to be added to the WF example. I suggest

-  .NET 6 is supported as the minimum requirement for the Dapr .NET SDK packages in this release. 
- Only .NET 8 and .NET 9 will be supported in Dapr v1.16 and later releases.

msfussell commented on daprdocs/content/en/getting-started/quickstarts/bindings-quickstart.md:
```diff
@@ -443,10 +443,18 @@ In the YAML file:
+
+{{% alert title="Note" color="primary" %}}
+
+Note that while .NET 6 is generally supported as the minimum .NET requirement across the Dapr .NET SDK packages
```
Same here and for the rest of the files. I would make this more succinct as below. Also the comment on Workflows only need to be added to the WF example. I suggest

.NET 6 is supported as the minimum requirement for the Dapr .NET SDK packages in this release.
Only .NET 8 and .NET 9 will be supported in Dapr v1.16 and later releases.

msfussell commented on daprdocs/content/en/getting-started/quickstarts/workflow-quickstart.md:
```diff
@@ -511,10 +511,18 @@ The `order-processor` console app starts and manages the lifecycle of an order p
+
+{{% alert title="Note" color="primary" %}}
+
+Note that while .NET 6 is generally supported as the minimum .NET requirement across the Dapr .NET SDK packages
```
I would make this more succinct as below and here add WF comment

.NET 6 is supported as the minimum requirement for the Dapr .NET SDK packages in this release.
- .NET 7 is the minimally supported version of .NET by Dapr.Workflows in Dapr v1.15
only 
Only .NET 8 and .NET 9 will be supported in Dapr v1.16 and later releases.

msfussell reviewed (commented):
See my review comments.

WhitWaldo committed:
Fixed recurring typo on last commit

Signed-off-by: Whit Waldo <whit.waldo@innovian.net>

msfussell commented:
@WhitWaldo - Did you see my comments here on making this into a set of bullet points with rewording?

WhitWaldo committed:
Making wording more concise, per suggestions.

Signed-off-by: Whit Waldo <whit.waldo@innovian.net>

WhitWaldo committed:
Minor dependency tweak

Signed-off-by: Whit Waldo <whit.waldo@innovian.net>

WhitWaldo committed:
Merge branch 'v1.15' into net9-sdk-quickstart-docs

WhitWaldo commented:
@msfussell Nope - missed them in the wall of notifications this morning, so thank you for pointing them out. Updates made throughout on the latest commits.

hhunter-ms commented on daprdocs/content/en/getting-started/quickstarts/actors-quickstart.md:
```diff
@@ -30,10 +30,17 @@ As a quick overview of the .NET actors quickstart:
 <!-- END_IGNORE -->
+- [.NET 6](https://dotnet.microsoft.com/download), [.NET 8](https://dotnet.microsoft.com/download) or [.NET 9](https://dotnet.microsoft.com/download) installed
+
+{{% alert title="Note" color="primary" %}}
```
@WhitWaldo `{{% alert ... %}}` shortcode typically doesn't render when put between `{{% codetab %}}` shortcode... could you[ build the site locally](https://github.com/dapr/docs?tab=readme-ov-file#run-local-server) and see if it renders for you?

WhitWaldo replied:
Oops - you caught me. Didn't test this one out locally. I'll see what I can get working on this.

hhunter-ms replied:
@WhitWaldo so far, we've used `> **Note:**` for notes within `codetab` shortcode. It does a soft block quote.

WhitWaldo replied:
I haven't had the time to circle back on this, but I'll try to get to it before the holidays!

WhitWaldo replied:
Removed the alert from the codetab and switched to a note

dependabot[bot] committed:
Bump nanoid from 3.3.2 to 3.3.8 in /daprdocs

Bumps [nanoid](https://github.com/ai/nanoid) from 3.3.2 to 3.3.8.
- [Release notes](https://github.com/ai/nanoid/releases)
- [Changelog](https://github.com/ai/nanoid/blob/main/CHANGELOG.md)
- [Commits](https://github.com/ai/nanoid/compare/3.3.2...3.3.8)

---
updated-dependencies:
- dependency-name: nanoid
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

hhunter-ms committed:
Merge branch 'v1.14' into dependabot/npm_and_yarn/daprdocs/nanoid-3.3.8

WhitWaldo committed:
Merge branch 'v1.15' into net9-sdk-quickstart-docs

hhunter-ms committed:
updates per elena pr for 1.15 (#4471)

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
Merge branch 'v1.14' into dependabot/npm_and_yarn/daprdocs/nanoid-3.3.8

hhunter-ms committed:
Merge pull request #4461 from dapr/dependabot/npm_and_yarn/daprdocs/nanoid-3.3.8

Bump nanoid from 3.3.2 to 3.3.8 in /daprdocs

hhunter-ms committed:
Merge branch 'v1.15' into net9-sdk-quickstart-docs

scott-brady committed:
feat(bindings-zeebe-command): Add support for passing variables to throw-error operation. (#3639)

Signed-off-by: Scott Brady <realscottbrady@gmail.com>

WhitWaldo committed:
Merge branch 'v1.15' into net9-sdk-quickstart-docs

yossisp committed:
Update workflow-features-concepts.md

Fix typos, use more Java-idiomatic code names (methods in Java usually don't begin with uppercase letters).

Signed-off-by: Yossi Spektor <easpex@gmail.com>

yossisp committed:
Update setup-postgresql-v2.md

Mention that `actorStateStore` needs to be `true` if the state store is used for workflows as well (since workflows use actors under the hood).

Signed-off-by: Yossi Spektor <easpex@gmail.com>

yossisp committed:
Update setup-postgresql-v1.md

Mention that `actorStateStore` needs to be `true` if the state store is used for workflows as well (since workflows use actors under the hood).

Signed-off-by: Yossi Spektor <easpex@gmail.com>

hhunter-ms committed:
Merge pull request #4478 from yossisp/patch-1

Update workflow-features-concepts.md

hhunter-ms committed:
Merge branch 'v1.14' into patch-2

hhunter-ms committed:
Merge branch 'v1.14' into patch-3

hhunter-ms committed:
Merge pull request #4480 from yossisp/patch-3

Update setup-postgresql-v1.md

hhunter-ms committed:
Merge branch 'v1.14' into patch-2

hhunter-ms committed:
Merge pull request #4479 from yossisp/patch-2

Update setup-postgresql-v2.md

hhunter-ms committed:
Merge branch 'v1.14' into feature/3639-zeebe-command-throw-error-variables

hhunter-ms committed:
Merge pull request #4477 from scott-brady/feature/3639-zeebe-command-throw-error-variables

feat(bindings-zeebe-command): Add support for passing variables to throw-error operation. (#3639)

hhunter-ms commented:
@WhitWaldo friendly bump - happy new year!

hhunter-ms committed:
Merge branch 'v1.15' into net9-sdk-quickstart-docs

Gallardot committed:
fix rust-sdk menu link

Signed-off-by: Gallardot <gallardot@apache.org>

hhunter-ms committed:
Merge pull request #4488 from Gallardot/Gallardot-patch-1

fix rust-sdk menu link

WhitWaldo committed:
Merge branch 'v1.15' into net9-sdk-quickstart-docs

WhitWaldo committed:
Merge branch 'dapr:v1.14' into net9-sdk-quickstart-docs

WhitWaldo committed:
Merge branch 'v1.15' into net9-sdk-quickstart-docs

WhitWaldo committed:
Fixed list numbering

Signed-off-by: Whit Waldo <whit.waldo@innovian.net>

WhitWaldo committed:
Fixed .NET SDK links

Signed-off-by: Whit Waldo <whit.waldo@innovian.net>

WhitWaldo committed:
Removed alert shortcode from within codetab block

Signed-off-by: Whit Waldo <whit.waldo@innovian.net>

WhitWaldo committed:
Updated all .NET SDK version links

Signed-off-by: Whit Waldo <whit.waldo@innovian.net>

WhitWaldo committed:
Updated all .NET package version notes to remove callout in favor of an inline note within the codetab blocks

Signed-off-by: Whit Waldo <whit.waldo@innovian.net>

WhitWaldo committed:
Merge branch 'v1.15' into net9-sdk-quickstart-docs

WhitWaldo commented:
@hhunter-ms It looks like it's one of your commits that failed DCO for some reason. Can you amend it?

hhunter-ms commented on daprdocs/content/en/getting-started/quickstarts/actors-quickstart.md:
```diff
@@ -30,10 +30,13 @@ As a quick overview of the .NET actors quickstart:
 <!-- IGNORE_LINKS -->
 - [Docker Desktop](https://www.docker.com/products/docker-desktop)
 <!-- END_IGNORE -->
+- [.NET 6](https://dotnet.microsoft.com/en-us/download/dotnet/6.0), [.NET 8](https://dotnet.microsoft.com/en-us/download/dotnet/8.0) or [.NET 9](https://dotnet.microsoft.com/en-us/download/dotnet/9.0) installed
```
```suggestion
- [.NET 6](https://dotnet.microsoft.com/download/dotnet/6.0), [.NET 8](https://dotnet.microsoft.com/download/dotnet/8.0) or [.NET 9](https://dotnet.microsoft.com/download/dotnet/9.0) installed
```

hhunter-ms commented on daprdocs/content/en/getting-started/quickstarts/bindings-quickstart.md:
```diff
@@ -443,10 +443,13 @@ In the YAML file:
 <!-- IGNORE_LINKS -->
 - [Docker Desktop](https://www.docker.com/products/docker-desktop)
 <!-- END_IGNORE -->
+- [.NET 6](https://dotnet.microsoft.com/en-us/download/dotnet/6.0), [.NET 8](https://dotnet.microsoft.com/en-us/download/dotnet/8.0) or [.NET 9](https://dotnet.microsoft.com/en-us/download/dotnet/9.0) installed
```
```suggestion
- [.NET 6](https://dotnet.microsoft.com/download/dotnet/6.0), [.NET 8](https://dotnet.microsoft.com/download/dotnet/8.0) or [.NET 9](https://dotnet.microsoft.com/download/dotnet/9.0) installed
```

hhunter-ms commented on daprdocs/content/en/getting-started/quickstarts/configuration-quickstart.md:
```diff
@@ -272,10 +272,13 @@ setTimeout(() => {
 <!-- IGNORE_LINKS -->
 - [Docker Desktop](https://www.docker.com/products/docker-desktop)
 <!-- END_IGNORE -->
+- [.NET 6](https://dotnet.microsoft.com/en-us/download/dotnet/6.0), [.NET 8](https://dotnet.microsoft.com/en-us/download/dotnet/8.0) or [.NET 9](https://dotnet.microsoft.com/en-us/download/dotnet/9.0) installed
```
```suggestion
- [.NET 6](https://dotnet.microsoft.com/download/dotnet/6.0), [.NET 8](https://dotnet.microsoft.com/download/dotnet/8.0) or [.NET 9](https://dotnet.microsoft.com/download/dotnet/9.0) installed
```

hhunter-ms commented on daprdocs/content/en/getting-started/quickstarts/pubsub-quickstart.md:
```diff
@@ -358,10 +358,13 @@ console.log("Published data: " + JSON.stringify(order));
 <!-- IGNORE_LINKS -->
 - [Docker Desktop](https://www.docker.com/products/docker-desktop)
 <!-- END_IGNORE -->
+- [.NET 6](https://dotnet.microsoft.com/en-us/download/dotnet/6.0), [.NET 8](https://dotnet.microsoft.com/en-us/download/dotnet/8.0) or [.NET 9](https://dotnet.microsoft.com/en-us/download/dotnet/9.0) installed
```
```suggestion
- [.NET 6](https://dotnet.microsoft.com/download/dotnet/6.0), [.NET 8](https://dotnet.microsoft.com/download/dotnet/8.0) or [.NET 9](https://dotnet.microsoft.com/download/dotnet/9.0) installed
```

hhunter-ms commented on daprdocs/content/en/getting-started/quickstarts/secrets-quickstart.md:
```diff
@@ -247,10 +247,13 @@ Order-processor output:
 <!-- IGNORE_LINKS -->
 - [Docker Desktop](https://www.docker.com/products/docker-desktop)
 <!-- END_IGNORE -->
+- [.NET 6](https://dotnet.microsoft.com/en-us/download/dotnet/6.0), [.NET 8](https://dotnet.microsoft.com/en-us/download/dotnet/8.0) or [.NET 9](https://dotnet.microsoft.com/en-us/download/dotnet/9.0) installed
```
```suggestion
- [.NET 6](https://dotnet.microsoft.com/download/dotnet/6.0), [.NET 8](https://dotnet.microsoft.com/download/dotnet/8.0) or [.NET 9](https://dotnet.microsoft.com/download/dotnet/9.0) installed
```

hhunter-ms commented on daprdocs/content/en/getting-started/quickstarts/serviceinvocation-quickstart.md:
```diff
@@ -315,10 +315,13 @@ console.log("Order passed: " + res.config.data);
 <!-- IGNORE_LINKS -->
 - [Docker Desktop](https://www.docker.com/products/docker-desktop)
 <!-- END_IGNORE -->
+- [.NET 6](https://dotnet.microsoft.com/en-us/download/dotnet/6.0), [.NET 8](https://dotnet.microsoft.com/en-us/download/dotnet/8.0) or [.NET 9](https://dotnet.microsoft.com/en-us/download/dotnet/9.0) installed
```
```suggestion
- [.NET 6](https://dotnet.microsoft.com/download/dotnet/6.0), [.NET 8](https://dotnet.microsoft.com/download/dotnet/8.0) or [.NET 9](https://dotnet.microsoft.com/download/dotnet/9.0) installed
```

hhunter-ms commented on daprdocs/content/en/getting-started/quickstarts/statemanagement-quickstart.md:
```diff
@@ -288,10 +288,13 @@ In the YAML file:
 <!-- IGNORE_LINKS -->
 - [Docker Desktop](https://www.docker.com/products/docker-desktop)
 <!-- END_IGNORE -->
+- [.NET 6](https://dotnet.microsoft.com/en-us/download/dotnet/6.0), [.NET 8](https://dotnet.microsoft.com/en-us/download/dotnet/8.0) or [.NET 9](https://dotnet.microsoft.com/en-us/download/dotnet/9.0) installed
```
```suggestion
- [.NET 6](https://dotnet.microsoft.com/download/dotnet/6.0), [.NET 8](https://dotnet.microsoft.com/download/dotnet/8.0) or [.NET 9](https://dotnet.microsoft.com/download/dotnet/9.0) installed
```

hhunter-ms commented on daprdocs/content/en/getting-started/quickstarts/workflow-quickstart.md:
```diff
@@ -507,10 +507,13 @@ The `order-processor` console app starts and manages the lifecycle of an order p
 <!-- IGNORE_LINKS -->
 - [Docker Desktop](https://www.docker.com/products/docker-desktop)
 <!-- END_IGNORE -->
+- [.NET 7](https://dotnet.microsoft.com/en-us/download/dotnet/7.0), [.NET 8](https://dotnet.microsoft.com/en-us/download/dotnet/8.0) or [.NET 9](https://dotnet.microsoft.com/en-us/download/dotnet/9.0) installed
```
```suggestion
- [.NET 7](https://dotnet.microsoft.com/download/dotnet/7.0), [.NET 8](https://dotnet.microsoft.com/download/dotnet/8.0) or [.NET 9](https://dotnet.microsoft.com/download/dotnet/9.0) installed
```

hhunter-ms reviewed (changes requested):
removing localization from urls

WhitWaldo committed:
Update daprdocs/content/en/getting-started/quickstarts/actors-quickstart.md

Co-authored-by: Hannah Hunter <94493363+hhunter-ms@users.noreply.github.com>
Signed-off-by: Whit Waldo <whit.waldo@innovian.net>

WhitWaldo committed:
Apply suggestions from code review

Co-authored-by: Hannah Hunter <94493363+hhunter-ms@users.noreply.github.com>
Signed-off-by: Whit Waldo <whit.waldo@innovian.net>

msfussell committed:
Merge branch 'v1.15' into net9-sdk-quickstart-docs

msfussell reviewed (approved):
LGTM
